### PR TITLE
jekyll: create plugin to fetch remote resources

### PR DESCRIPTION
**Pull request by crazy-max**
~needs #14685~

create a Jekyll plugin to fetch remote resources so they will be available as preview in Netlify when PR are created. also adds another plugin to update api toc.

### CODE REVIEW

**Comment by crazy-max** [comment]
logs: https://app.netlify.com/sites/docsdocker/deploys/62760653ca87f2000865d66e#L186

image

fetched resource available in preview: https://deploy-preview-14688--docsdocker.netlify.app/registry/configuration/

**Comment by crazy-max** [comment]
added another commit to replace update-api-toc script with a jekyll plugin so we can see results in preview too. Before:

image

After:

image

**Comment by crazy-max** [comment]
As discussed with @thaJeztah, moved move the JSON config for the fetch_remote plugin to the `_config.yml` ofr better visibility.

File: _config.yml [hunk 1]
```diff
@@ -72,3 +84,63 @@ defaults:
...
+#   - dest is the destination path within the working tree
+#   - src is a list of glob source paths within the remote repository
+fetch-remote:
+  - repo: docker/cli
+    ref: refs/heads/20.10
+    paths:
+      - dest: engine/extend
+        src:
+          - "docs/extend/**"
+          - "!docs/extend/cli_plugins.md" # the cli plugins api is not a stable API, and not included in the TOC for that reason.
```
Comment (usha-mandya) [inline _config.yml:101]: The `cli_plugins.md` file doesnt exist in the upstream repo anymore. Maybe we can remove this line? https://github.com/docker/cli/tree/20.10/docs/extend

Comment (crazy-max) [reply _config.yml:101]: Good point, I just blindly translated the old shell script.

**Review by usha-mandya — APPROVED** [review]

**Comment by crazy-max** [comment]
@thaJeztah As discussed updated the config yml with full repo uri as well as simplified git ref.

**Review by thaJeztah — APPROVED** [review]
LGTM!

**Comment by thaJeztah** [comment]
YOLO! Let's get this one merged!

**Pull request merged**